src/lsm/forest_fuzz.zig

```diff
@@ -0,0 +1,418 @@
+const std = @import("std");
+const testing = std.testing;
+const allocator = testing.allocator;
+const assert = std.debug.assert;
+const os = std.os;
+
+const config = @import("../config.zig");
+const vsr = @import("../vsr.zig");
+const log = std.log.scoped(.lsm_forest_fuzz);
+
+const MessagePool = @import("../message_pool.zig").MessagePool;
+const Transfer = @import("../tigerbeetle.zig").Transfer;
+const Account = @import("../tigerbeetle.zig").Account;
+const Storage = @import("../test/storage.zig").Storage;
+const StateMachine = @import("../state_machine.zig").StateMachineType(Storage);
+
+const GridType = @import("grid.zig").GridType;
+const GrooveType = @import("groove.zig").GrooveType;
+const Forest = StateMachine.Forest;
+
+const Grid = GridType(Storage);
+const SuperBlock = vsr.SuperBlockType(Storage);
+
+const FuzzOp = union(enum) {
+    // TODO Test checkpointing and range queries too.
+    compact,
+    put: Account,
+    get: Get,
+
+    const Get = union(enum) {
+        id: u128,
+        // TODO Support querying secondary indices too.
+    };
+};
+
+const Environment = struct {
+    const cluster = 32;
+    const replica = 4;
+    // TODO Is this appropriate for the number of fuzz_ops we want to run?
+    const size_max = vsr.Zone.superblock.size().? + vsr.Zone.wal.size().? + 1024 * 1024 * 1024;
+
+    const node_count = 1024;
+    // This is the smallest size that set_associative_cache will allow us.
+    const cache_entries_max = 2048;
+    const forest_options = StateMachine.forest_options(.{
+        // Ignored by StateMachine.forest_options().
+        .lsm_forest_node_count = undefined,
+        .cache_entries_accounts = cache_entries_max,
+        .cache_entries_transfers = cache_entries_max,
+        .cache_entries_posted = cache_entries_max,
+        .message_body_size_max = config.message_size_max - @sizeOf(vsr.Header),
+    });
+
+    const State = enum {
+        uninit,
+        init,
+        formatted,
+        superblock_open,
+        forest_open,
+        forest_compacting,
+    };
+
+    state: State,
+    dir_fd: os.fd_t,
+    fd: os.fd_t,
+    storage: *Storage,
+    message_pool: MessagePool,
+    superblock: SuperBlock,
+    superblock_context: SuperBlock.Context,
+    grid: Grid,
+    forest: Forest,
+    // We need @fieldParentPtr() of forest, so we can't use an optional Forest.
+    forest_exists: bool = false,
```

Comment:
```suggestion
    forest_exists: bool,
```

Since `Environment` is initially `undefined`, the unused default value might be misleading.